xswap/preprocessing.py

```diff
@@ -0,0 +1,106 @@
+import pathlib as _pathlib
+
+
+def load_str_edges(filename, node_delim=',', edge_delim='\n'):
+    """
+    Load edges from file into memory. Store edges as a list and store each edge
+    as Tuple[str, str]. Used to load edges for preprocessing.
+    """
+    file_path = _pathlib.Path(filename)
+    if not file_path.is_file():
+        raise FileNotFoundError
+
+    with open(file_path, 'r') as f:
+        graph_string = f.read()
+    str_edges = [
+        edge.split(node_delim) for edge in graph_string.split(edge_delim) if edge != ''
+    ]
+    return str_edges
+
+
+def load_processed_edges(filename):
+    """
+    Load processed edges from a file. Processed means that edges are guaranteed
+    to be integers ranging from zero to the number of unique nodes.
+    """
+    str_edges = load_str_edges(filename)
+    edges = [
+        (int(edge[0]), int(edge[1])) for edge in str_edges
+    ]
+    return edges
+
+
+def _map_nodes_to_int(nodes):
+    """
+    Return a dict mapping a list of nodes to their sorted indices. Nodes should
+    be a list of strings.
+
+    Returns:
+    --------
+    Dict[str, int]
+    """
+    name_to_id = dict()
+    sorted_node_set = sorted(list(set(nodes)))
+    for i, name in enumerate(sorted_node_set):
+        name_to_id[name] = i
+    return name_to_id
+
+
+def _apply_map(edges, mapping):
+    """
+    Maps edges according to a node mapping. If `mapping` contains two dicts,
+    `_apply_map` maps source and target nodes separately. Else, if `mapping`
+    contains only one dict, then source and target nodes are mapped according
+    to the same mapping.
+    """
```

Comment:
This seems a little hacky. Would it make sense for the function to take both source_map and target_map? Then the calling function could just pass the same dict to source_map and target_map in the case they are of the same metanode.